packages/client/bin/cli.ts

```diff
@@ -783,15 +805,54 @@ async function run() {
   initKZG(kzg, args.trustedSetup ?? __dirname + '/../src/trustedSetups/official.txt')
   // Give network id precedence over network name
   const chain = args.networkId ?? args.network ?? Chain.Mainnet
+  const cryptoFunctions: CustomCrypto = {}
+  if (args.useJsCrypto === false) {
+    await waitReady()
+    cryptoFunctions.keccak256 = keccak256WASM
+    cryptoFunctions.ecrecover = (
+      msgHash: Uint8Array,
+      v: bigint,
+      r: Uint8Array,
+      s: Uint8Array,
+      chainID?: bigint
+    ) =>
+      secp256k1Expand(
+        secp256k1Recover(
+          msgHash,
+          concatBytes(setLengthLeft(r, 32), setLengthLeft(s, 32)),
+          Number(calculateSigRecovery(v, chainID))
+        )
+      ).slice(1)
+    cryptoFunctions.sha256 = wasmSha256
+    cryptoFunctions.ecsign = (msg: Uint8Array, pk: Uint8Array, chainId?: bigint) => {
```

Comment:
Why is this?  We don't want the wasm crypto dep to be in any other package than `client` so we don't mess up library consumers who have challenges with WASM.  Every wrapper function will likely be custom since the underlying crypto functions may have different signatures.

Reply:
Ah ok, this makes sense :smile: